### PR TITLE
Add 'kitty' image backend

### PR DESCRIPTION
**Pull request by SolitudeSF**
## Description

Added [kitty](https://github.com/kovidgoyal/kitty) terminal image output.

### CODE REVIEW

File: neofetch [hunk 1]
```diff
@@ -2979,11 +2979,16 @@ make_thumbnail() {
     image_name="$crop_mode-$crop_offset-$width-$height-${image##*/}"
 
     # Handle file extensions.
-    case "${image##*.}" in
-        "eps"|"pdf"|"svg"|"gif"|"png")
-            image_name+=".png" ;;
-        *)  image_name+=".jpg" ;;
-    esac
+    if [ $image_backend == 'kitty' ]; then
```
Comment (konimex) [inline neofetch]: Note that bashism is preferred so use `[[` instead of `[`.

**Comment by Snuggle** [comment]
My gosh, I would love to see this merged. +1
Thanks for the PR, @SolitudeSF! :heart:

**Comment by dylanaraps** [comment]
Merged manually with some changes. https://github.com/dylanaraps/neofetch/commit/2606e4b1d6c431efc2a9bea4f0bec2385ad805f2

Thanks!

**Comment by SolitudeSF** [comment]
Kitty can display png images directly. If you supply a jpg or other format it uses imagemagick to convert it to png, so its additional startup overhead, is there no way to include special case for kitty?

**Comment by dylanaraps** [comment]
Neofetch doesn’t convert the image, it just appends png to the filename

**Comment by SolitudeSF** [comment]
in understand, but you call imagemagick anyway to generate thumbnail. or am i misunderstanding something?

**Comment by dylanaraps** [comment]
Yes, `neofetch` generates a thumbnail. We do this to speed up the display of images. Neofetch mostly deals with wallpaper sized files so this is necessary to avoid 1s+ run times.

The lines of code you added around the `case` statement don't actually do anything and they don't skip the thumbnail generation.

**Comment by SolitudeSF** [comment]
yes, they dont skip, they force thumbnail to `png` format when kitty is used to avoid additional convertation call from kitty's side.

**Comment by dylanaraps** [comment]
This sounds like an issue with kitty more so than neofetch. Why should we add a workaround when it can be fixed directly?

**Comment by SolitudeSF** [comment]
thats not an issue, thats how it works, it can display png data directly. i just optimize one `convert` call away.

**Comment by SolitudeSF** [comment]
actually, after benchmarking both methods, my *workaround* consistently yields slower result, which i dont understand, but at least this issue is finally closed.

**Pull request closed without merging**